packages/tasit-action/src/contract/Action.js

```diff
@@ -6,32 +6,67 @@ import ConfigLoader from "../ConfigLoader";
 //  and/or MetaTxAction subclasses
 export class Action extends Subscription {
   #provider;
+  #signer;
+  #rawTx;
   #tx;
   #txConfirmations;
   #timeout;
   #lastConfirmationTime;
 
-  constructor(txPromise, provider) {
+  constructor(rawTx, provider, signer) {
```

Comment:
This is the line I should have commented on. Moving this https://github.com/tasitlabs/TasitSDK/pull/357/files/ea53439d35f7f5e022c6d3e1e4f09d6b4b02b617#r279085297 thread here.

Reply:
> Hmm, the `Action` is receiving the `rawTx` and the `signer`, on the `send()` function (it's an alias for `signAndSend()` the `rawTx` is being decorated (gasParam and nonce), sign and sent.

That makes sense that it's signed after gasParam and nonce, but I'm glad you reminded me of that in this context.

> We can have the action receiving the `signedTx` instead but will need to handle with gas and nonce outside?

Hmm, I wasn't thinking of moving gas and nonce outside. I was thinking of assuming who the signer is whenever possible outside the scope of action creation. So maybe already setting it for the `tasit-account` and then the action uses that. Half-baked thought, though.

> A question: Shouldn't all transactions (even the meta-tx) be signed before broadcasted?

Yes, they definitely need to be signed by an account before being broadcast to a relayer or the Ethereum network. I guess I could imagine an app powered the Tasit SDK building an action that is then serialized and deep-linked over to a mobile wallet app that then broadcasts it to a relayer or the Ethereum network, though. But we can build around such a feature when the time comes, not preemptively

Reply:
I think we can move the signer from the constructor to a setter function and check it that's assigned on the `send` function.

Reply:
Would that mean that we'd have to set the signer for every action as a separate step after building it? I think we'd only want to go that route if we could use that setter on `tasit-account` and then actions would use that signer.

Reply:
Hmm, this change (signer from constructor to setter) will be a tiny change for now, since we are using `Contract` to create actions, this change will be almost transparent. But could reflect better how the `Action` behavior should be (Allows creation of an Action to be signed later).

Note: The current `Contract` code will throw an error if a write function be called without a signer/wallet set. Maybe we should change that check to the `Action.send` function.

Could you clarify that sentence? 
>  if we could use that setter on tasit-account and then actions would use that signer.

Reply:
That all sounds good to me.

>> if we could use that setter on tasit-account and then actions would use that signer.

>Could you clarify that sentence?

I guess I actually meant using the `tasit-account` account as the signer in `Contract` within `tasit-action`

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/372

Reply:
Resolving this thread so we can continue the discussion in that new issue